### PR TITLE
Lots of fixes

### PR DESCRIPTION
**Pull request by TheAlienKnight**
All changes were tested *except* for autosync auto deletion for templates and servers. I'll need someone to verify that works, it should though. 

Changes:
- Edited a number of queries that were being run for counts, to only attempt to fetch the estimated mongodb count, vs actually fetching the items, putting them into an array, and checking the length. 
- Changed the interval in which "daily" metrics postings occurred, to actually be a day (23.8h to be exact) and not be posting every **5 seconds**. This was... causing a lot of problems
- Changed how the site launches, it was possible for it to become "stuck" if it happened to execute quickly- it would end up waiting for the bot to emit ready. but it never got to the point where the bot could emit ready, as it held the process up.
- If the datadog variable is an empty string, or null in the settings.json file, don't attempt to log metrics
- Changed a few files to no longer call **all** databases, even when there is a switch operator to control what metrics need to be updated. Ie: Don't waste time and resources querying things that aren't being used currently
- For automatic banlist updating, and this will only impact servers with no bans, or development testing- if there are no bans to be found, don't attempt to set a null value in redis
- For express, the cookie-session module wants an array of strings now, even though it isn't mentioned in the docs, I tried it a few times. I just tossed the settings string into an array to make it happy.

TLDR; The site shouldn't hang and die anymore, at least it didn't in my testing. I also am assuming any memory leak that did exist, has been fixed with these changes (from what I can see anyway, and the stats, this is correct).

### CODE REVIEW

**Comment by IceeMC** [comment]
bark

**Comment by ConnorDoesDev** [comment]
bork

**Review by IceeMC — APPROVED** [review]
pog

**Review by advaith1 — CHANGES_REQUESTED** [review]
for autosync deletion can you make it only delete on the specific error codes for invalid invite/invalid template? we don't want to nuke everything on outages and stuff

**Comment by advaith1** [comment]
also awaiting getters is weird; is there realistically much of a chance that the guilds/channels arent cached during operation? did you run into that?

can we just retrieve from cache and use optional chaining while accessing, so it doesn't totally break if they arent cached?

**Comment by TheAlienKnight** [comment]
> also awaiting getters is weird; is there realistically much of a chance that the guilds/channels arent cached during operation? did you run into that?

I actually ran into an issue where since my test guild is small with only a few test accounts, it wasn't actually caching the members. I thought it was odd, and the only explanation I have is that perhaps discord.js doesn't attempt to cache smaller amounts of members if they aren't actively online/interacting. I changed it for that reason, I honestly agree it is weird, I wasn't expecting it to be the issue, but it ended up being one that impacted me during testing. Regardless, it doesn't hurt to have that fallback either- even if for a guild the size of DEL it doesn't happen often. There's always a better way, but I was trying to work around the existing code vs implement new methods/ways of doing things

> can we just retrieve from cache and use optional chaining while accessing, so it doesn't totally break if they arent cached?

Possibly? I can mess around with it and see it, I honestly just went with what made the most immediate sense/worked to solve the issues I ran into

> for autosync deletion can you make it only delete on the specific error codes for invalid invite/invalid template? we don't want to nuke everything on outages and stuff

I can yes, I didn't think about outages actually, good point/catch there

**Comment by TheAlienKnight** [comment]
Also copying the summary here:

- No longer uselessly awaiting getters (should only happen [once if needed](https://discord.js.org/#/docs/discord.js/v13/class/Channel?scrollTo=fetch))
- Only delete templates and servers on proper [opcodes]( https://discord.com/developers/docs/topics/opcodes-and-status-codes#json)
  - Server: 10006
  - Template: 10057

**Review by Carolina2k22 — APPROVED** [review]

File: src/Routes/autosync.ts [hunk 1]
```diff
@@ -215,7 +256,52 @@ router.get('/templates', async (req, res) => {
...
+            .collection("templates")
+            .deleteOne({ _id: id });
+
+        await global.db.collection("audit").insertOne({
+            type: "REMOVE_TEMPLATE",
+            executor: "AutoSync",
+            target: id,
+            date: Date.now(),
+            reason: "Unknown server template (10057)",
+            reasonType: 0 // since this is the only option i guess? - AJ
```
Comment (Carolina2k22) [inline src/Routes/autosync.ts:271]: `reasonType` should have the value of `4`.

File: src/Routes/autosync.ts [hunk 2]
```diff
@@ -159,7 +160,47 @@ router.get('/servers', async (req, res) => {
...
-        discord.channels.alerts.send(`${settings.emoji.warn} failed to autosync server **${server.name}** \`(${id})\`: ${e}\n<${settings.website.url}/servers/${id}>`)
+        if (e.code != 10006) return; // https://discord.com/developers/docs/topics/opcodes-and-status-codes#json
+        await global.db.collection("servers").deleteOne({ _id: id });
+        await global.db.collection("audit").insertOne({
+            type: "REMOVE_SERVER",
+            executor: "AutoSync",
+            target: id,
+            date: Date.now(),
+            reason: "Failed to autosync server, assuming the invite is invalid.",
+            reasonType: 0 // again, assuming 0 is the only option here
```
Comment (Carolina2k22) [inline src/Routes/autosync.ts:171]: `reasonType` should have the value of `5`.

**Review by Carolina2k22 — CHANGES_REQUESTED** [review]
everything else seems okay

**Comment by Carolina2k22** [comment]
just re whether your issues have been addressed advaith

**Review by Carolina2k22 — APPROVED** [review]

**Comment by TheAlienKnight** [comment]
The changes I just pushed fix the audit logs trying to use require() in ESM, it also fixes the crash that occurs when bots are offline on startup. Those are the last two things @advaith1 pointed out in the discord.

**Review by Carolina2k22 — APPROVED** [review]
